packages/frontend/src/hooks/useKeycard.ts

```diff
@@ -1,27 +1,83 @@
-import { useLocalStorage } from "@uidotdev/usehooks";
+import { useAccount, useWalletClient } from "wagmi";
 import { generatePrivateKey, privateKeyToAccount } from "viem/accounts";
+import { getLogger } from "@logtape/logtape";
+import { useQuery, useQueryClient } from "@tanstack/react-query";
+
+import { RelayClient } from "@massmarket/client";
+import { getWindowLocation } from "@massmarket/utils";
 
 import { useShopId } from "./useShopId.ts";
 import { KeycardRole } from "../types.ts";
+import { useRelayEndpoint } from "./useRelayEndpoint.ts";
+
+const logger = getLogger(["mass-market", "frontend", "useKeycard"]);
 
-export function useKeycard(
-  keycard?: {
-    privateKey: `0x${string}`;
-    role: KeycardRole;
-    address: `0x${string}`;
-  },
-) {
-  if (!keycard) {
-    const privateKey = generatePrivateKey();
-    const account = privateKeyToAccount(privateKey);
-    keycard = {
-      privateKey,
-      role: KeycardRole.NEW_GUEST,
-      address: account.address,
-    };
-  }
+/**
+ * This hook returns a keycard for the browser session.
```

Comment:
i would be more explicit here, the hook return a keycard for a shop given a role which currently is either a "merchant" or "guest" from tanstack's cache.